src/modules/dotnet-publish.ts

```diff
@@ -27,40 +40,75 @@ const DotnetPublish = {
         return `Publishes the dotnet solution from the root of the project (via ${this.cmd()})`;
     },
     getOptions(): OptionStringBuilder {
-        return new OptionStringBuilder("publish", "p");
+        return Options.Publish;
     },
     /**
      * Runs a publish of the dotnet solution to the local file system
      * @param {string} absoluteOutputDir Optional absolute path of release output directory. If not provided,
      * defaults to dotnet solution's 'release' directory
      */
     run(absoluteOutputDir?: string) {
+        DotnetPath.solutionPathOrExit();
+
         if (StringUtils.isEmpty(absoluteOutputDir)) {
             absoluteOutputDir = DotnetPath.releaseDir();
         }
 
-        Echo.message(`Cleaning release directory '${absoluteOutputDir}'...`);
-        shell.rm("-rf", absoluteOutputDir!);
-        Echo.success(" - Successfully cleaned released directory");
-        Echo.newLine();
+        _cleanReleaseDirOrExit(absoluteOutputDir!);
```

Comment:
I'm noticing a trend `...Exit(`.  Wasn't aware we were naming things like this.  I ask the question then whether the exit is erroring or not.  That bit is unclear.

Reply:
It's a pattern we started with some of the shared modules where we were knew we needed the value returned from the function and the program/command couldn't continue without it

So instead of doing it manually in the module everytime, ie:

```ts
const solutionPath = DotnetPath.solutionPath();

if (solutionPath == null) {
    Echo.error("Unable to find dotnet solution file");
    shell.exit(1)
}
```

that logic is bundled up in the `solutionPathOrExit()` method

For something like these private functions to encapsulate some of the cleaning/building logic that aren't really meant to be shared, the `OrExit` part could probably be dropped - it can safely be assumed that they will run until an error is encountered which would (should) cause the program to exit.